added initial code

kingofthebongo2008 opened the pull request:
project builds

added references only to the third party git

added clang and claude

reformatted

applied clang-format

replaced unique ptr with shader ptr

kingofthebongo2008 committed:
added initial code

project builds

added references only to the third party git

added clang and claude

reformatted

applied clang-format

replaced unique ptr with shader ptr

itsafuu commented:
thirdparty should not be a submodule of the project. https://github.com/GeniusVentures/cmaketemplate should be a submodule as the directory "build". 

cmaketemplate will download thirdparty artifacts from github and extract to one level above this repo directory. If you want to compile thirdparty yourself you can simply clone it to one level above this repo, cmaketemplate will find it.

kingofthebongo2008 commented:
ok, fixing

On Wed, 27 Aug 2025 at 18:49, itsafuu ***@***.***> wrote:

> *itsafuu* left a comment (GeniusVentures/SGInterCommunication#1)
> <https://github.com/GeniusVentures/SGInterCommunication/pull/1#issuecomment-3228751136>
>
> thirdparty should not be a submodule of the project.
> https://github.com/GeniusVentures/cmaketemplate should be a submodule as
> the directory "build".
>
> cmaketemplate will download thirdparty artifacts from github and extract
> to one level above this repo directory. If you want to compile thirdparty
> yourself you can simply clone it to one level above this repo,
> cmaketemplate will find it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GeniusVentures/SGInterCommunication/pull/1#issuecomment-3228751136>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIXS4NZVOGT43HLDHULDQT3PXHRPAVCNFSM6AAAAACE5Z5NFOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTEMRYG42TCMJTGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kingofthebongo2008 committed:
removed dependency

fix submodule

Auto stash before rebase of "initial-setup"

remove cmake lists

fixed issues with build

kingofthebongo2008 committed:
Auto stash before rebase of "initial-setup"

kingofthebongo2008 committed:
reworked project

kingofthebongo2008 committed:
added reworked files

kingofthebongo2008 committed:
updated files

kingofthebongo2008 committed:
simplified code with protobuf

kingofthebongo2008 committed:
WIP on initial-setup

remove dead code

kingofthebongo2008 committed:
libp2p stub code

kingofthebongo2008 committed:
remove define

kingofthebongo2008 committed:
remove std cout

kingofthebongo2008 committed:
removed cerr

kingofthebongo2008 committed:
std::filesystem

kingofthebongo2008 committed:
fix port negotiation example

kingofthebongo2008 committed:
add multi instance example

kingofthebongo2008 committed:
enable libp2p
